gms/impl/src/main/java/com/linkedin/metadata/resources/ml/MLModels.java

```diff
@@ -0,0 +1,285 @@
+                value.setMlModelProperties(modelProperties);
+            } else if (aspect instanceof Ownership) {
+                Ownership ownership = Ownership.class.cast(aspect);
+                value.setOwnership(ownership);
+            } else if (aspect instanceof QuantitativeAnalyses) {
+                QuantitativeAnalyses quantitativeAnalyses = QuantitativeAnalyses.class.cast(aspect);
+                value.setQuantitativeAnalyses(quantitativeAnalyses);
+            } else if (aspect instanceof SourceCode) {
+                SourceCode sourceCode = SourceCode.class.cast(aspect);
+                value.setSourceCode(sourceCode);
+            } else if (aspect instanceof Status) {
+                Status status = Status.class.cast(aspect);
+                value.setStatus(status);
+            } else if (aspect instanceof TrainingData) {
+                TrainingData trainingData = TrainingData.class.cast(aspect);
+                value.setTrainingData(trainingData);
+            }
+        });
+        return value;
+    }
+
+    private static MLModelFactors combineFactors(MLModelFactorPrompts mlModelFactorPrompts) {
+        MLModelFactors mlModelFactors = new MLModelFactors();
+        StringArray environment = new StringArray();
+        StringArray groups = new StringArray();
+        StringArray instrumentation = new StringArray();
+        if (mlModelFactorPrompts.hasEvaluationFactors()) {
+            for (MLModelFactors modelFactors : mlModelFactorPrompts.getEvaluationFactors()) {
+                if (modelFactors.hasEnvironment()) {
+                    environment.addAll(modelFactors.getEnvironment());
+                }
+                if (modelFactors.hasGroups()) {
+                    groups.addAll(modelFactors.getGroups());
+                }
+                if (modelFactors.hasInstrumentation()) {
+                    instrumentation.addAll(modelFactors.getInstrumentation());
+                }
+            }
+        }
+        if (mlModelFactorPrompts.hasRelevantFactors()) {
+            for (MLModelFactors modelFactors : mlModelFactorPrompts.getRelevantFactors()) {
+                if (modelFactors.hasEnvironment()) {
+                    environment.addAll(modelFactors.getEnvironment());
+                }
+                if (modelFactors.hasGroups()) {
+                    groups.addAll(modelFactors.getGroups());
+                }
+                if (modelFactors.hasInstrumentation()) {
+                    instrumentation.addAll(modelFactors.getInstrumentation());
+                }
+            }
+        }
+        mlModelFactors.setEnvironment(environment);
+        mlModelFactors.setGroups(groups);
+        mlModelFactors.setInstrumentation(instrumentation);
+        return mlModelFactors;
+    }
+
+    @Nonnull
+    @Override
+    protected MLModelSnapshot toSnapshot(@Nonnull MLModel mlModel, @Nonnull MLModelUrn urn) {
+        return null;
```

Comment:
curious, why is this not returning `MlModelSnapshot` object? Also the method signature says `Nonnull` but this is returning a null object.

Reply:
Sorry forgot to implement it. This method doesn't appear to be used so it didn't impact any of my testing. I'll add a similar implementation to the equivalent classes.